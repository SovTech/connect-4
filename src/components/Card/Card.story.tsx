import * as React from 'react';
import { storiesOf } from '@storybook/react';
import Card from '../Card';
import ContainerWrapper from '../ContainerWrapper';

storiesOf('Card', module)
  .add('All', () => (
    <ContainerWrapper>
      <Card
        noPadding={true}
        contentPadding={true}
        title='Title'
      >
        <div>With some content</div>
      </Card>
    </ContainerWrapper>
  ));
