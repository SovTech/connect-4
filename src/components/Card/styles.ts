import theme from '../../theme';
import styled from 'styled-components';

export const CardWrapper = styled.div`
    background: #fff;
    border: 1px solid ${theme.overlayColor};
    border-radius: ${theme.borderRadius};
    box-shadow: 0 1px 2px 0 ${theme.overlayColor};
    margin: ${theme.marginHalf};
    max-width: 100%;
    min-width: 300px;
`;

export const CardInner = styled.div`
    height: 100%;
`;

export const PaddedContent = styled.div`
    padding: ${theme.marginHalf};
`;

export const TitleWrapper = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    color: ${theme.inverseTextColor};
    border-bottom: 1px solid ${theme.overlayColor};
    padding: 10px 20px;
    border-top-left-radius: ${theme.borderRadius};
    border-top-right-radius: ${theme.borderRadius};

    * {
      margin: 0;
    }
`;