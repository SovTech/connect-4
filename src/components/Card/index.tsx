// Card

import * as React from 'react';
import TimeAgo from 'react-timeago';
import { H3 } from '../HeaderText';
import { CardInner, CardWrapper, PaddedContent, TitleWrapper } from './styles';

type Props = {
  children?: any;
  style?: React.CSSProperties;
  date?: string;
  loading?: boolean;
  titleLoader?: boolean;
  noPadding?: boolean;
  contentPadding?: boolean;
  greyBG?: boolean;
  title?: string;
  onClick?: Function;
  className?: string;
}

export default (props: Props) => (
  <CardWrapper style={props.style}>
    <CardInner>
      {props.title
        ? <TitleWrapper>
          {props.loading && props.titleLoader ? <div /> : <H3>{props.title}</H3>}
          {props.date ? <TimeAgo date={props.date} /> : <div />}
        </TitleWrapper>
        : null}
      {props.contentPadding ? <PaddedContent>{props.children}</PaddedContent> : props.children}
    </CardInner>
  </CardWrapper>
)
