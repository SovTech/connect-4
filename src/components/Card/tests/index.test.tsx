import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Card from '../index';

describe('Card Component', () => {
  it('renders a Card correctly', () => {
    const tree = renderer
      .create((
        <Card
          title='Card Title'
          onClick={() => console.log('onClick')}
        />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a Card correctly when loading', () => {
    const tree = renderer
      .create((
        <Card
          loading={true}
          title='Card Title'
          onClick={() => console.log('onClick')}
        />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a Card correctly when loading with a title loading indicator', () => {
    const tree = renderer
      .create((
        <Card
          loading={true}
          noPadding={true}
          contentPadding={true}
          greyBG={true}
          title='Card Title'
          onClick={() => console.log('onClick')}
        />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a Card correctly with no title padding, content padding and a grey background', () => {
    const tree = renderer
      .create((
        <Card
          noPadding={true}
          contentPadding={true}
          greyBG={true}
          title='Card Title'
          onClick={() => console.log('onClick')}
        />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
