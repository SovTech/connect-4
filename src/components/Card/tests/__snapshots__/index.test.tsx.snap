// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders a Card correctly 1`] = `
<div
  className="sc-ifAKCX kiXPab"
  style={undefined}
>
  <div
    className="sc-EHOje cqsmXB"
  >
    <div
      className="sc-gzVnrw hwiGRg"
    >
      <h3
        className="sc-htpNat gLVzZr"
      >
        Card Title
      </h3>
      <div />
    </div>
  </div>
</div>
`;

exports[`Card Component renders a Card correctly when loading 1`] = `
<div
  className="sc-ifAKCX kiXPab"
  style={undefined}
>
  <div
    className="sc-EHOje cqsmXB"
  >
    <div
      className="sc-gzVnrw hwiGRg"
    >
      <h3
        className="sc-htpNat gLVzZr"
      >
        Card Title
      </h3>
      <div />
    </div>
  </div>
</div>
`;

exports[`Card Component renders a Card correctly when loading with a title loading indicator 1`] = `
<div
  className="sc-ifAKCX kiXPab"
  style={undefined}
>
  <div
    className="sc-EHOje cqsmXB"
  >
    <div
      className="sc-gzVnrw hwiGRg"
    >
      <h3
        className="sc-htpNat gLVzZr"
      >
        Card Title
      </h3>
      <div />
    </div>
    <div
      className="sc-bZQynM gRvvsu"
    />
  </div>
</div>
`;

exports[`Card Component renders a Card correctly with no title padding, content padding and a grey background 1`] = `
<div
  className="sc-ifAKCX kiXPab"
  style={undefined}
>
  <div
    className="sc-EHOje cqsmXB"
  >
    <div
      className="sc-gzVnrw hwiGRg"
    >
      <h3
        className="sc-htpNat gLVzZr"
      >
        Card Title
      </h3>
      <div />
    </div>
    <div
      className="sc-bZQynM gRvvsu"
    />
  </div>
</div>
`;
